build/xtask/src/dist.rs

```diff
@@ -808,6 +814,12 @@ fn check_rebuild(toml: &Config) -> Result<()> {
     Ok(())
 }
 
+#[derive(Debug, Hash)]
```

Comment:
What's the purpose of the hash impl on this type? Will it be factored into image ID?

If yes, I'm concerned about including a machine-specific path in it.

Reply:
This isn't changing anything; I'm just moving the data structure from `hubtools` back into Hubris proper, to simplify `hubtools`.

The `Hash` derivation is unchanged from before, e.g. [here's a commit predating `hubtools`](https://github.com/oxidecomputer/hubris/blob/f2bcd883b084c0481d3608c9e48209ac72e71360/build/xtask/src/dist.rs#L775-L779).

This _is_ included in image ID (and always have been), which is seeded with
```rust
fn build_kernel(
    cfg: &PackageConfig,
    allocs: &Allocations,
    all_output_sections: &mut BTreeMap<u32, LoadSegment>,
    all_memories: &IndexMap<String, Range<u32>>,
    entry_points: &HashMap<String, u32>,
    image_name: &str,
    secure: &Option<SecureData>,
) -> Result<(u32, BTreeMap<String, u32>)> {
    let mut image_id = fnv::FnvHasher::default();
    all_output_sections.hash(&mut image_id);
```
so we're not any worse than before!